fix(aws-ecs): refactor high level aws-ecs constructs into the aws-ecs-patterns module

pkandasamy91 opened the pull request:
Refactored all high level constructs from the aws-ecs module into the aws-ecs-patterns module as discussed offline. I've left the LoadBalancesXxxService constructs in the aws-ecs module as I think it make more sense to leave them there. 

BREAKING CHANGE: These changes move all L3 and higher constructs out of the aws-ecs module into the aws-ecs-patterns module. The following constructs have been moved into the aws-ecs-patterns module: EcsQueueWorkerService, FargateQueueWorkerService, LoadBalancedEcsService, LoadBalancedFargateService and LoadBalancedFargateServiceApplets. 

### Pull Request Checklist

* [X] Testing
  - Unit test added (prefer not to modify an existing test, otherwise, it's probably a breaking change)
  - __CLI change?:__ coordinate update of integration tests with team
  - __cdk-init template change?:__ coordinated update of integration tests with team
* [X] Docs
  - __jsdocs__: All public APIs documented
  - __README__: README and/or documentation topic updated
  - __Design__: For significant features, design document added to `design` folder
* [X] Title and Description
  - __Change type__: title prefixed with **fix**, **feat** and module name in parens, which will appear in changelog
  - __Title__: use lower-case and doesn't end with a period
  - __Breaking?__: last paragraph: "BREAKING CHANGE: <describe what changed + link for details>"
  - __Issues__: Indicate issues fixed via: "**Fixes #xxx**" or "**Closes #xxx**"
* [ ] Sensitive Modules (requires 2 PR approvers)
  - IAM Policy Document (in @aws-cdk/aws-iam)
  - EC2 Security Groups and ACLs (in @aws-cdk/aws-ec2)
  - Grant APIs (only if not based on official documentation with a reference)

----

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license.

Doug-AWS commented:
@awslabs/aws-cdk-team Are we going to follow this model? Do we want to come up with what we call these? CDK Pattern Resources?

rix0rrr commented:
I would call this a **fix** and it needs a BREAKING CHANGE notice for the classes you moved.

pkandasamy91 commented:
> I would call this a **fix** and it needs a BREAKING CHANGE notice for the classes you moved.

Updated the title to be fix, and added a BREAKING CHANGE notice.
